docs/topics/multiplatform/multiplatform-hierarchy.md

```diff
@@ -166,167 +106,146 @@ kotlin {
+1. Reapply the template by explicitly calling `applyDefaultHierarchyTemplate()`.
+2. Configure additional source sets [manually](#manual-configuration) using `dependsOn()`:
 
-<tabs group="build-script">
-<tab title="Kotlin" group-key="kotlin">
+    <tabs group="build-script">
+    <tab title="Kotlin" group-key="kotlin">
 
-```kotlin
-kotlin {
-    iosX64()
-    iosArm64()
+    ```kotlin
+    kotlin {
+        jvm()
+        macosArm64()
+        iosArm64()
+        iosSimulatorArm64()
     
-    sourceSets {
-        val commonMain by getting
-        val iosX64Main by getting
-        val iosArm64Main by getting
-        val iosMain by creating {
-            dependsOn(commonMain)
-            iosX64Main.dependsOn(this)
-            iosArm64Main.dependsOn(this)
+        // Apply the default hierarchy again. It'll create, for example, the iosMain source set:
+        applyDefaultHierarchyTemplate()
+    
+        sourceSets {
+            // Create an additional jvmAndMacos source set:
+            val jvmAndMacos by creating {
+                dependsOn(commonMain.get())
+            }
+    
+            macosArm64Main.get().dependsOn(jvmAndMacos)
+            jvmMain.get().dependsOn(jvmAndMacos)
         }
     }
-}
-```
+    ```
 
-</tab>
-<tab title="Groovy" group-key="groovy">
+    </tab>
+    <tab title="Groovy" group-key="groovy">
 
-```groovy
-kotlin {
-    iosX64()
-    iosArm64()
+    ```groovy
+    kotlin {
+        jvm()
+        macosArm64()
+        iosArm64()
+        iosSimulatorArm64()
     
-    sourceSets {
-        iosMain {
-            dependsOn(commonMain)
-            iosX64Main.dependsOn(it)
-            iosArm64Main.dependsOn(it)
-        }
+        // Apply the default hierarchy again. It'll create, for example, the iosMain source set:
+        applyDefaultHierarchyTemplate()
+    
+        sourceSets {
+            // Create an additional jvmAndMacos source set:
+            jvmAndMacos {
+                dependsOn(commonMain.get())
+            }
+            jvmAndMacos  {
+                dependsOn(macosArm64Main.get())
+            }
+            jvmAndMacos  {
+                dependsOn(jvmMain.get())
+            }
+        } 
     }
-}
-```
+    ```
 
-</tab>
-</tabs>
+    </tab>
+    </tabs>
 
-#### Target shortcuts and ARM64 (Apple Silicon) simulators
+#### Modifying source sets
 
-The `ios`, `watchos`, and `tvos` target shortcuts don't include the simulator targets for ARM64 (Apple Silicon)
-platforms: `iosSimulatorArm64`, `watchosSimulatorArm64`, and `tvosSimulatorArm64`. If you use the target shortcuts
-and want to build the project for an Apple Silicon simulator, make the following adjustment to the build script:
+**Case**. You already have the source sets with the exact same names as those generated by the template but shared among
+different sets of targets in your project. For example, a `nativeMain` source set shared among not all the native
```

Comment:
Will it be better if we rephrase?

For example, a `nativeMain` source set is shared among only specific native targets: `linuxX64`, `mingwX64`, and `macosX64`.